plugin-api/src/main/java/org/sonar/api/security/ExternalUsersProvider.java

```diff
@@ -42,19 +44,46 @@ public UserDetails doGetUserDetails(Context context) {
 
   public static final class Context {
     private String username;
+    /**
+     * @deprecated since 9.16 use {@link #httpRequest} instead
+     */
+    @Deprecated(since = "9.16", forRemoval = true)
     private HttpServletRequest request;
```

Comment:
this is a private field, not really an API that needs to be deprecated. It's not like we can replace its use.